### ISSUE TITLE
[Bug]: OneSignal.initWithContext() crashes on (5.0.0) 

### ISSUE THREAD

**Issue by SonicOctavesSuyash** [OneSignal/OneSignal-Android-SDK#1812]

**Comment by khurramaleee** [comment]
Did you find any solution?

**Comment by suyashraikar** [comment]
Still not any solution @khurramaleee

**Comment by michael-winkler** [comment]
@emawby Can you please take a look here too?

**Comment by emawby** [comment]
@michael-winkler Yep we are trying to reproduce this one as well is this only happening on Amazon fire devices?

**Comment by suyashraikar** [comment]
@michael-winkler No this is happening on all devices.

**Comment by bordot** [comment]
Having the same issue.

Exception java.lang.RuntimeException:

 ```
 at android.app.ActivityThread.callActivityOnStop (ActivityThread.java:5843)
  at android.app.ActivityThread.performStopActivityInner (ActivityThread.java:5809)
  at android.app.ActivityThread.handleStopActivity (ActivityThread.java:5880)
  at android.app.servertransaction.StopActivityItem.execute (StopActivityItem.java:43)
  at android.app.servertransaction.ActivityTransactionItem.execute (ActivityTransactionItem.java:45)
  at android.app.servertransaction.TransactionExecutor.executeLifecycleState (TransactionExecutor.java:176)
  at android.app.servertransaction.TransactionExecutor.execute (TransactionExecutor.java:97)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:2574)
  at android.os.Handler.dispatchMessage (Handler.java:106)
  at android.os.Looper.loopOnce (Looper.java:226)
  at android.os.Looper.loop (Looper.java:313)
  at android.app.ActivityThread.main (ActivityThread.java:8762)
  at java.lang.reflect.Method.invoke
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:604)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1067)
Caused by java.lang.NullPointerException: null cannot be cast to non-null type com.amazon.device.iap.PurchasingListener
  at com.onesignal.core.internal.purchases.impl.TrackAmazonPurchase.onUnfocused (TrackAmazonPurchase.kt:124)
  at com.onesignal.core.internal.application.impl.ApplicationService$handleLostFocus$1.invoke (ApplicationService.kt:325)
  at com.onesignal.core.internal.application.impl.ApplicationService$handleLostFocus$1.invoke (ApplicationService.kt:325)
  at com.onesignal.common.events.EventProducer.fire (EventProducer.kt:44)
  at com.onesignal.core.internal.application.impl.ApplicationService.handleLostFocus (ApplicationService.kt:325)
  at com.onesignal.core.internal.application.impl.ApplicationService.onActivityStopped (ApplicationService.kt:184)
  at android.app.Application.dispatchActivityStopped (Application.java:510)
  at android.app.Activity.dispatchActivityStopped (Activity.java:1548)
  at android.app.Activity.onStop (Activity.java:2679)
  at androidx.fragment.app.FragmentActivity.onStop (FragmentActivity.java:496)
  at androidx.appcompat.app.AppCompatActivity.onStop (AppCompatActivity.java:252)
  at android.app.Instrumentation.callActivityOnStop (Instrumentation.java:1551)
  at android.app.Activity.performStop (Activity.java:8829)
  at android.app.ActivityThread.callActivityOnStop (ActivityThread.java:5835)
```

The crashes seem to have been from Android users that installed from Play Store, not Amazon store. Samsung devices, not Amazon devices also.

**Comment by brismithers** [comment]
@suyashraikar could you post which version of `com.amazon.device:amazon-appstore-sdk` you're including as a dependency for your app?

**Comment by suyashraikar** [comment]
@brismithers we don't include any "com.amazon.device:amazon-appstore-sdk" any specific dependency in app. I guess this is internally used by OneSignal

**Comment by brismithers** [comment]
Good news, I was able to reproduce the issue with `com.amazon.device:amazon-appstore-sdk:3.0.3`, the crash happens if an unfocus happens early during initialization, I believe we need to add a little bit more to the fix we provided in #1860.  Will get that fix out shortly!

@suyashraikar I don't believe this issue can happen unless your app (either explicitly or through a dependency of yours) loads in this library, the OneSignal SDK depends on it as [compileOnly](https://github.com/OneSignal/OneSignal-Android-SDK/blob/user-model/main/OneSignalSDK/onesignal/core/build.gradle#L66) and has a [runtime check](https://github.com/OneSignal/OneSignal-Android-SDK/blob/user-model/main/OneSignalSDK/onesignal/core/src/main/java/com/onesignal/core/internal/purchases/impl/TrackAmazonPurchase.kt#L226) to ensure this code only runs when the library is loaded.  To be sure I've covered everything, can you run `./gradlew app:dependencies` and let me know the version of `com.amazon.device:amazon-appstore-sdk`?

**Comment by jennantilla** [comment]
Hello everyone! Just a heads up that this should be resolved in SDK versions >=[5.0.4](https://github.com/OneSignal/OneSignal-Android-SDK/releases). Please feel free to update to the latest version and let us know if there are any additional questions or concerns!
